### PR TITLE
Refactor glotaran.model

### PR DESCRIPTION
**Pull request by s-weigand**
Restructured model package, not to overwrite modules with members ([see](https://github.com/glotaran/pyglotaran/pull/444#issuecomment-733197804)) and fixed a circular dependency issue caused by it.

This reduces the type checking errors from [231](https://github.com/glotaran/pyglotaran/pull/444/checks?check_run_id=1453400975) to [199](https://github.com/s-weigand/pyglotaran/runs/1453403849?check_suite_focus=true).

### CODE REVIEW

File: glotaran/parse/register.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 """A register for models"""
-import typing
+from __future__ import annotations
 
-from glotaran.model.model import Model
+from typing import TYPE_CHECKING
```
Comment (jsnel) [inline glotaran/parse/register.py:4]: Looks a bit odd, but I guess if it's a standard pattern, just go for it.

File: glotaran/parse/register.py [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+
+if TYPE_CHECKING:
+    from typing import List
+
+    from glotaran.model import Model
```
Comment (jsnel) [inline glotaran/parse/register.py:9]: Looks a bit odd, but I guess if it's a standard pattern, just go for it.

**Review by jsnel — APPROVED** [review]
Looks good. Improves readability of the code for me.

File: glotaran/parse/register.py (hunk 2 above)
Comment (s-weigand) [reply glotaran/parse/register.py:9]: Since the import is only used for type checking, it kinda makes sense to only import it when actually type checking.

**Comment by joernweissenborn** [comment]
I approve under the condition that @jsnel  will do the rebase of the refactor/optimization branch.

**Review by joernweissenborn — APPROVED** [review]

**Pull request merged**